visualization/app/codeCharta/state/store/metricData/metricData.splitter.ts

```diff
@@ -0,0 +1,20 @@
+import { RecursivePartial, CCAction, MetricData, EdgeMetricData } from "../../../codeCharta.model"
+
+// Plop: Append action splitter import here
+import { splitEdgeMetricDataAction } from "./edgeMetricData/edgeMetricData.splitter"
+import { splitNodeMetricDataAction } from "./nodeMetricData/nodeMetricData.splitter"
+
+export function splitMetricDataActions(payload: RecursivePartial<MetricData>): CCAction[] {
+	const actions: CCAction[] = []
+
+	// Plop: Append action split here
+	if (payload.edgeMetricData !== undefined) {
+		actions.push(splitEdgeMetricDataAction(payload.edgeMetricData as EdgeMetricData[]))
+	}
+
+	if (payload.nodeMetricData !== undefined) {
```

Comment:
Could the payload contain both, `edgeMetricData` and `nodeMetricData` at the same time?

Reply:
Yes. We added this functionality to be able to set the state partially. 
You could call e.g.  `storeService.dispatch(setMetricData( { nodeMetricData: [], edgeMetricData: [] } ))`. The payload is propagated to it's subreducer. In this case, nodeMetricData will be forwarded to the `nodeMetricData.reducer`.

But the really cool functionality here is, that you can also call it with a partial state, e.g. `storeService.dispatch(setMetricData( { nodeMetricData: [] } ))` would also be a valid call. It would be the same as calling `storeService.dispatch(setNodeMetricData([]))`
This becomes really handy when you want to update multiple properties of the state at once. We do this, when we apply the scenarios.